base/linalg/diagonal.jl

```diff
@@ -234,6 +234,11 @@ A_mul_B!(out::AbstractMatrix, A::Diagonal, in::AbstractMatrix) = out .= A.diag .
 Ac_mul_B!(out::AbstractMatrix, A::Diagonal, in::AbstractMatrix) = out .= ctranspose.(A.diag) .* in
 At_mul_B!(out::AbstractMatrix, A::Diagonal, in::AbstractMatrix) = out .= transpose.(A.diag) .* in
 
+# ambiguities with Symmetric/Hermitian
+A_mul_Bt(A::Diagonal, B::RealHermSymComplexSym) = A*B
+At_mul_B(A::RealHermSymComplexSym, D::Diagonal) = A*B
+A_mul_Bc(A::Diagonal, B::RealHermSymComplexHerm) = A*B
+Ac_mul_B(A::RealHermSymComplexHerm, D::Diagonal) = A*B
```

Comment:
> Are these definitions correct (considering the recursive nature of `[c]transpose`)?

Turns out it is correct after all -- definition of the alias `RealHermSymComplexHerm` only allows element types `<:Number` so we will never dispatch here when the element type is a matrix for example (and thus this should not affect the behavior of block matrices). Probably worth adding a comment.